src/components/utils/string.js

```diff
@@ -0,0 +1,26 @@
+import { identity, compose } from './func'
+import { cleanFalsy, flatten } from './array'
+
+const firstUpper = str => str.substr(0, 1).toUpperCase() + str.substr(1)
+
+const lower = str => str.toLowerCase()
+
+const unCamel = str => str.replace(/([A-Z])/g, ' $1').split(' ')
+
+const compactFlat = compose(cleanFalsy, flatten)
+
+const asWords = str =>
+  compactFlat(((str || '').match(/[^\s_\-]+/g) || []).map(unCamel)).map(lower)
```

Comment:
Minor: how come flattening is necessary?
Minor: is it possible to extract the regex and name it as a constant? they are hard to read, hence why a named constant helps.
Minor: isn't it better to use a default variable instead of the `foo | ''` patern?

Reply:
> how come flattening is necessary?
After matching by `' '`, `'_'` and `'-'`, it will split camel case words on each match. So it might end up with nested arrays

> is it possible to extract the regex and name it as a constant? they are hard to read, hence why a named constant helps.
Absolutely, will do

> isn't it better to use a default variable instead of the foo | '' pattern
Default argument value would only cover the `undefined` case, with `|| ''` I'm trying to cover `null` values too